### ISSUE TITLE
Microsoft.Web/sites/config/web ipSecurityRestrictions

### ISSUE THREAD

**Issue by slavizh** [Azure/arm-template-whatif#82]
## Describe the noise

**Resource type** (i.e. Microsoft.Storage/storageAccounts)
Microsoft.Web/sites/config

**apiVersion** (i.e. 2019-04-01)
2019-08-01

**Client (PowerShell, Azure CLI, or API)**
PowerShell

**Relevant ARM Template code (we only need the resource object for the above `resourceType` and `apiVersion`, but if it's easier you can include the entire template**

**Expected response (i.e. "I expected no noise since the template has not been modified since the resources were deployed)**
No change

**Current (noisy) response (either include a screenshot of the what-if output, or copy/paste the text)**

```
~ Microsoft.Web/sites/app3434ssfdfsdf/config/web [2019-08-01]
    ~ properties.ipSecurityRestrictions: [
      - 1:

          action:      "Deny"
          description: "Deny all access"
          ipAddress:   "Any"
          name:        "Deny all"
          priority:    2147483647

      ]
    ~ properties.scmIpSecurityRestrictions: [
      ~ 0:

        - tag: "Default"

      - 1:

          action:      "Deny"
          description: "Deny all access"
          ipAddress:   "Any"
          name:        "Deny all"
          priority:    2147483647

      ]


```

**Additional context**

**Comment by Xynoclafe** [comment]
@slavizh Can you give us the template that was used to get this noise? It'll help us figure out the noise better.

**Comment by slavizh** [comment]
@Xynoclafe yes but can you give me your MSFT e-mail so I can send them in private. I cannot share them publicly.

**Comment by torreymicrosoft** [comment]
Thanks for your patience, this was recently picked up as one of the older open issues for the ARM Deployments team to review. Apologies for the six years, and for the fact that the request for your template back in 2020 does not appear to have gone anywhere after you offered to share it privately.

The good news is that I do not need the template to answer it. What-If for Azure Deployment Stacks went generally available today with noise reduction enabled in all regions: a stack records a what-if baseline when it is deployed, and any property that is unchanged between that baseline and the current evaluation is removed from the result as a false positive.

Your output shows two App Service behaviours rather than a bug in your template. The rule at index 1 with `action: "Deny"`, `name: "Deny all"`, `ipAddress: "Any"` and `priority: 2147483647` is the implicit terminal deny rule App Service appends to `properties.ipSecurityRestrictions` and `properties.scmIpSecurityRestrictions` once any allow rule exists. The `- tag: "Default"` line on `scmIpSecurityRestrictions[0]` is the default rule tag the service stamps on. I confirmed with Bicep 0.46.1 that `tag` really is a declarable property on `Microsoft.Web/sites/config@2024-04-01`, so this is server populated data rather than something you got wrong. Neither is in your template, both come back on GET, and both are identical on every run, which is exactly what the baseline filters.

You reported this from PowerShell, so at resource group scope:

```
New-AzResourceGroupDeploymentStackWhatIfResult `
  -Name whatif-preview `
  -ResourceGroupName <resource-group> `
  -StackResourceId <deployment-stack-resource-id> `
  -TemplateFile main.bicep `
  -ActionOnUnmanage DetachAll `
  -DenySettingsMode None `
  -RetentionInterval PT3H
```

One honest caveat before you try it. The baseline is recorded when the stack is deployed or updated, so deploy or update the stack once to establish it, then run what-if against the stack. The docs are also clear that noise reduction removes many of these differences but not every one, so it is still worth reading the result rather than assuming everything left is real.

Announcement: https://techcommunity.microsoft.com/blog/azuregovernanceandmanagementblog/now-generally-available-what-if-for-azure-deployment-stacks/4547614
Docs: https://learn.microsoft.com/azure/azure-resource-manager/bicep/deployment-stacks-what-if

I am closing this out. If the `Deny all` rule or the `tag` line still shows up in a Stacks What-if result once the baseline is in place, reopen here or file in `Azure/deployment-stacks` with the what-if result resource ID.
